lib/request-handler.js

```diff
@@ -153,15 +155,31 @@ class RequestHandler {
     if (this.requestOptions.captureStackTrace) {
       Error.captureStackTrace(this.stackContainer = {});
     }
+    this._callback = callback;
     const self = this;
-    this.loadBalancingPolicy.newQueryPlan(this.client.keyspace, this.requestOptions, function newPlanCb(err, iterator) {
-      if (err) {
-        return callback(err);
-      }
-      self._hostIterator = iterator;
-      self._callback = callback;
+    if (this.requestOptions.host) {
+      // if host is configured bypass load balancing policy and use
+      // a single host plan.
+      let hostProvided = false; 
+      self._hostIterator = { 
```

Comment:
We can reuse existing `utils.arrayIterator([ host ])` or you can write a small generator function (Node.js 4 supports it):

```javascript
function* singleHostIterator(h) {
  yield h;
}

self._hostIterator = singleHostIterator(host);
```

Reply:
:+1: that's much better, will change it.